src/http/adapter.js

```diff
@@ -68,6 +68,14 @@ function checkPermissionFor( state, user, context, action ) {
 		.then( onPermission, onError );
 }
 
+function executeStack( resource, envelope, stack ) {
+	var calls = stack.slice( 0 );
+	var call = calls.shift();
+	return call ?
+		call.bind( resource )( envelope, executeStack.bind( undefined, resource, envelope, calls ) ) :
+		{ status: 500, data: { message: 'The server failed to provide a response' } };
+}
```

Comment:
Do we need a way to short circuit the stack if middleware returns something? Meaning, what should be the effect if middleware calls `next` and then returns something? Right now it looks like the stack will continue to execute but the first returned value will be used as the response. Honestly not sure what would be most correct.